packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/ExpoNotificationBuilder.java

```diff
@@ -205,4 +216,53 @@ private int getPriority() {
     // By default let's show the notification
     return NotificationCompat.PRIORITY_HIGH;
   }
+
+  /**
+   * The method first tries to get the icon from the manifest's meta-data {@link #META_DATA_DEFAULT_ICON_KEY}.
+   * If a custom setting is not found, the method falls back to using app icon.
+   *
+   * @return Resource ID for icon that should be used as a notification icon.
+   */
+  protected int getIcon() {
+    try {
+      ApplicationInfo ai = getContext().getPackageManager().getApplicationInfo(getContext().getPackageName(), PackageManager.GET_META_DATA);
+      if (ai.metaData.containsKey(META_DATA_DEFAULT_ICON_KEY)) {
+        return ai.metaData.getInt(META_DATA_DEFAULT_ICON_KEY);
+      }
+    } catch (Exception e) {
```

Comment:
I know it is common pattern across expo codebase, but I dislike catching such general error.

I assume this is to catch `PackageManager.NameNotFoundException`? Maybe we should consider catching it explicitly and rethrowing? It should never ever happen and only scenario I can think of it occurring  is when someone makes some unintentional change in here.